packages/common/src/TransactionUtils.ts

```diff
@@ -101,17 +132,32 @@ export const runTransaction = async ({
     const maximumGasPriceMultiple = 2 * 3;
     // Pre-London transactions require `gasPrice`, London transactions require `maxFeePerGas` and `maxPriorityFeePerGas`
 
-    let receipt;
+    let receipt: any;
```

Comment:
As mentioned above, it could make sense to set this receipt type to `TransactionReceipt | PromiEvent<TransactionReceipt>`.

Reply:
Why can't we make this `TransactionReceipt | PromiEvent<TransactionReceipt>` since it's going to be used as the return value?

Reply:
this is added now